libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFiltersConfiguration.ts

```diff
@@ -39,23 +36,47 @@ export function useDependentDateFiltersConfiguration(
         filterElementsByDate,
     );
 
+    const originalCommonDateFilterState = useDependentCommonDateFilterConfigurationState(commonDateFilter);
+
     const [dependentDateFilters, setDependentDateFilters] =
         useState<IDashboardDependentDateFilter[]>(originalState);
 
-    const onDependentDateFiltersSelect = (dataSet: ObjRef, isSelected: boolean) => {
-        invariant(!isUriRef(dataSet));
+    const [dependentCommonDateFilter, setDependentCommonDateFilter] = useState<IDashboardDateFilter>(
+        originalCommonDateFilterState,
+    );
+
+    const onDependentDateFiltersSelect = (
+        item: IDashboardDependentDateFilter,
+        isSelected: boolean,
+        isCommonDate?: boolean,
+    ) => {
+        const localIdentifier = item.dataSet;
+        invariant(!isUriRef(localIdentifier));
 
         const changedParentIndex = dependentDateFilters.findIndex(
-            (dependentDateFilter) => dependentDateFilter.localIdentifier === dataSet.identifier,
+            (dependentDateFilter) =>
+                dependentDateFilter.localIdentifier === localIdentifier?.identifier &&
+                dependentDateFilter.isCommonDate === isCommonDate,
         );
-        const changedItem = { ...dependentDateFilters[changedParentIndex] };
 
-        changedItem.isSelected = isSelected;
+        if (changedParentIndex === -1) {
+            setDependentDateFilters([...dependentDateFilters, item]);
+        } else {
+            if (isCommonDate) {
+                const dependentDateFiltersState = dependentDateFilters;
+                dependentDateFiltersState.splice(changedParentIndex, 1);
```

Comment:
These array mutation functions should not be used as they introduce various subtle code bugs. For example here, you assigned  `dependentDateFilters` to the new variable but still mutated the original array you are iterating over IMHO. Remaping of the array would be better.

Reply:
Replaced with filter.